src/shell/atuin.zsh

```diff
@@ -54,6 +55,6 @@ if [[ -z $ATUIN_NOBIND ]]; then
 	bindkey '^r' _atuin_search_widget
 
 	# depends on terminal mode
-	bindkey '^[[A' _atuin_search_widget
-	bindkey '^[OA' _atuin_search_widget
+	bindkey '^[[A' _atuin_search_widget --shell-up-key-binding
+	bindkey '^[OA' _atuin_search_widget --shell-up-key-binding
```

Comment:
We actually need to do a little more to get this working for ZSH

First define a new function

```
_atuin_up_search(){
  _atuin_search --shell-up-key-binding
}
```

Then bind a new widget

```
zle -N _atuin_up_search_widget _atuin_up_search
```

then bind the new widget instead

```
	bindkey '^[[A' _atuin_up_search_widget
	bindkey '^[OA' _atuin_up_search_widget
```

With those changes this works great on ZSH! Thanks for the work here 🚀

Reply:
Maybe you can push those changes directly to this branch?

Reply:
Sure, I've just done that! I won't normally do that by default as it can be seen as a little rude 🙈

Reply:
I'm personally fine with it, makes the collaboration better IMO.

When on the other side, I usually submit GitHub suggestions so the PR author can accept the change or not.

Reply:
>  I usually submit GitHub suggestions

Likewise, but the UI doesn't tend to let you submit suggestions outside of the diff :/ Or at least I couldn't get it to

Reply:
Indeed.